Packs/TrendMicroEmailSecurity/Integrations/TrendMicroEmailSecurityEventCollector/TrendMicroEmailSecurityEventCollector.py

```diff
@@ -0,0 +1,488 @@
+
+
+""" HELPER FUNCTIONS """
+
+
+def add_missing_fields_to_event(event: dict):
+    for field in (
+        "action",
+        "mailID",
+        "sender",
+        "genTime",
+        "logType",
+        "subject",
+        "tlsInfo",
+        "senderIP",
+        "direction",
+        "eventType",
+        "messageID",
+        "recipient",
+        "domainName",
+        "headerFrom",
+        "policyName",
+        "eventSubtype",
+        "policyAction",
+        "deliveredTo",
+        "attachments",
+        "recipients",
+        "headerTo",
+        "details",
+        "timestamp",
+        "size",
+        "deliveryTime",
+        "reason",
+        "embeddedUrls",
+    ):
+        if field not in event:
+            event[field] = None
+
+
+def convert_datetime_to_without_milliseconds(time_: str) -> str:
+    return time_.strip("Z").split(".")[0] + "Z"
+
+
+class Deduplicate:
+    def __init__(self, last_ids_fetched: list[str], event_type: EventType) -> None:
+        self.is_fetch_time_moved: bool = False
+        self.new_event_ids_suspected: list[str] = []
+        self.last_event_ids_suspected: list[str] = last_ids_fetched
+        self.event_type: EventType = event_type
+
+    def get_last_time_event(self, events: list[dict]) -> str:
+        """
+        Extract the latest time from the events list
+        """
+
+        latest_time_event = max(
+            events,
+            key=lambda event: datetime.strptime(
+                convert_datetime_to_without_milliseconds(event["genTime"]),
+                DATE_FORMAT_EVENT,
+            ),
+        )
+        return convert_datetime_to_without_milliseconds(latest_time_event["genTime"])
+
+    def update_suspected_duplicate_events_list(self, event: dict, start: str):
+        """
+        Adds the ID of the event as long as genTime is equal to start
+        """
+        if not self.is_fetch_time_moved and (
+            convert_datetime_to_without_milliseconds(event["genTime"]) == start
+        ):
+            self.new_event_ids_suspected.append(self.generate_id_for_event(event))
+        else:
+            self.is_fetch_time_moved = True
+            self.new_event_ids_suspected = []
+
+    def get_events_with_duplication_risk(
+        self, events: list[dict], latest_time: str
+    ) -> list[dict]:
+        """
+        Returns all events whose genTime is equal to the latest time
+        from the list of events returned from the API
+        """
+        return list(
+            filter(
+                lambda event: convert_datetime_to_without_milliseconds(event["genTime"])
+                == latest_time,
+                events,
+            )
+        )
+
+    def generate_id_for_event(self, event: dict) -> str:
+        """
+        Generate an ID from mailID value for mail tracking type or from messageID for policy type.
+        """
+        if self.event_type == EventType.POLICY_LOGS:
+            return event["messageID"]
```

Comment:
we are not adding a prefix to the Ids for the type? are we sure messageID will never equal mailID?